### DISCUSSION TITLE
[Question] Should we remove the back button from Avo?

### DISCUSSION THREAD

**Discussion by adrianthedev** [avo-hq/avo#2031] [Feature feedback]
Hey everyone, we're thinking of dropping the back button from Avo's Show, New, and Edit views.

The reasoning behind this is that we jump through many hoops to ensure the back button redirects back to where we originally came from.

One of the things we do is send many `params` in the URL to detect the previous page.
Sometimes we send those params through form submissions (record updates) so that we have them on the redirect.

- We can't use the `javascript:back` method because it's inconsistent when we submit forms.
- The `request.referrer` variable doesn't always help as sometimes it's just a link

I love the back button, and I use it every day. 
It's just the maintenance is getting ridiculous.

Open to ideas and feedback.

![CleanShot 2023-11-15 at 09 45 39@2x](https://github.com/avo-hq/avo/assets/1334409/cd6b1cce-045b-474a-90d7-b082e144c357)

**Comment by fractaledmind** [comment]
Is the behavior of this button intended to be different from the browser's back button in certain key situations?

**Reply by adrianthedev** [reply]
It is sometimes.
Here's one scenario:

- go to the users index page
- select a user and land on a user's show page
- hit edit and land on the edit page
- his save and get redirected to the show page

When using the browser back button, you'd be redirected to the form. 
When using our button you'd be redirected to the users index page.

**Reply by adrianthedev** [reply]
THe `javascript:back` event also doesn't cover scenarios where the user just navigated to a URL and there's not history in the browser :|

**Comment by robzolkos** [comment]
If its something truly useful and being used then should probably keep it.  37signals wrote how they handled it https://dev.37signals.com/better-navigation-in-hey/

**Reply by adrianthedev** [reply]
Good article outlining the issues with such a "smart" back button. Thank you Rob!

**Comment by adrianthedev** [comment]
After a few more conversation about it on [Discord](https://discord.com/channels/740892036978442260/1174020779931869264) we have decided that the back button will live on.
It provides contextual value to users.

Thanks everyone for your feedback!
